Documentation/gettingstarted/cilium_install_crio.rst

```diff
@@ -0,0 +1,119 @@
+      secret "cilium-etcd-secrets" created
+      daemonset.extensions "cilium" created
+      clusterrolebinding.rbac.authorization.k8s.io "cilium" created
+      clusterrole.rbac.authorization.k8s.io "cilium" created
+      serviceaccount "cilium" created
+
+  .. group-tab:: K8s 1.9
+
+    .. parsed-literal::
+
+      $ kubectl create -f \ |SCM_WEB|\/examples/kubernetes/1.9/cilium-crio.yaml
+      configmap "cilium-config" created
+      secret "cilium-etcd-secrets" created
+      daemonset.extensions "cilium" created
+      clusterrolebinding.rbac.authorization.k8s.io "cilium" created
+      clusterrole.rbac.authorization.k8s.io "cilium" created
+      serviceaccount "cilium" created
+
+  .. group-tab:: K8s 1.10
+
+    .. parsed-literal::
+
+      $ kubectl create -f \ |SCM_WEB|\/examples/kubernetes/1.10/cilium-crio.yaml
+      configmap "cilium-config" created
+      secret "cilium-etcd-secrets" created
+      daemonset.extensions "cilium" created
+      clusterrolebinding.rbac.authorization.k8s.io "cilium" created
+      clusterrole.rbac.authorization.k8s.io "cilium" created
+      serviceaccount "cilium" created
+
+  .. group-tab:: K8s 1.11
+
+    .. parsed-literal::
+
+      $ kubectl create -f \ |SCM_WEB|\/examples/kubernetes/1.11/cilium-crio.yaml
+      configmap "cilium-config" created
+      secret "cilium-etcd-secrets" created
+      daemonset.extensions "cilium" created
+      clusterrolebinding.rbac.authorization.k8s.io "cilium" created
+      clusterrole.rbac.authorization.k8s.io "cilium" created
+      serviceaccount "cilium" created
+
+Kubernetes is now deploying Cilium with its RBAC settings, ConfigMap
+and DaemonSet as a pod on minikube. This operation is performed in the
+background.
+
+Run the following command to check the progress of the deployment:
+
+::
+
+    $ kubectl get daemonsets -n kube-system
+    NAME         DESIRED   CURRENT   READY     UP-TO-DATE   AVAILABLE   NODE SELECTOR   AGE
+    cilium       1         1         0         1            1           <none>          3m
+    kube-proxy   1         1         1         1            1           <none>          8m
+
+Wait until the cilium Deployment shows a ``CURRENT`` count of ``1``
+like above (a ``READY`` value of ``0`` is OK for this tutorial).
+
+Since CRI-O does not automatically detect a new CNI plugin installed you will
+need to restart CRI-O daemon to pick the Cilium CNI configuration.
+
+First make sure cilium is running.
+
+::
+
+    $ kubectl get pods -n kube-system -o wide
+    NAME               READY     STATUS    RESTARTS   AGE       IP          NODE
+    cilium-mqtdz       1/1       Running   0          3m       10.0.2.15   minikube
+
+After that you can restart CRI-O:
+
+::
+
+    $ minikube ssh -- sudo systemctl restart crio
+
+And finally, you need to restart Cilium pod so it can re-mount the
+``/var/run/crio.sock`` recreated by CRI-O inside the new Cilium pod.
+
+::
+
+    $ kubectl delete -n kube-system pod cilium-mqtdz
```

Comment:
It might be good to explain this bit a little more, perhaps with a note box (so it's clear that it isn't an instruction). It isn't clear that there is a circular dependency here, where both cilium and crio only do something on startup.

Separately, deleting the pod will uninstall the cni config, right? but this works because crio isn't also restarted during this time. Correct?

Reply:
@raybejjani 

>It might be good to explain this bit a little more, perhaps with a note box (so it's clear that it isn't an instruction). It isn't clear that there is a circular dependency here, where both cilium and crio only do something on startup.

Since CRI-O does not automatically detect that a new CNI plugin has been
installed, you will need to restart the CRI-O daemon for it to pick up the
Cilium CNI configuration.

Better?

>Separately, deleting the pod will uninstall the cni config, right? but this works because crio isn't >also restarted during this time. Correct?

Correct

Reply:
Yes. I did also mean the part about cilium needing a restart but it would be confusing to include it in that paragraph.